icechunk-python/src/lib.rs

```diff
@@ -432,6 +464,25 @@ impl PyIcechunkStore {
         Ok(snapshot_id.to_string())
     }
 
+    fn list_refs<'py>(
```

Comment:
can we also provide `async_list_refs`?

Reply:
can we add a python test that exercises this new function?

Reply:
Yes I can do both. I'd like some input on https://github.com/earth-mover/icechunk/pull/318#discussion_r1813703484 first before I tackle those.